receiver/elasticsearchreceiver/internal/model/nodestats.go

```diff
@@ -0,0 +1,164 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/elasticsearchreceiver/internal/model"
+
+// NodeStats represents a response from elasticsearch's /_nodes/stats endpoint.
+// The struct is not exhaustive; It does not provide all values returned by elasticsearch,
+// only the ones relevant to the metrics retrieved by the scraper.
+type NodeStats struct {
```

Comment:
I appreciate the readability this brings but one concern I always have with this kind of structural representation is that we may end up losing the ability to differentiate between missing values and zero values. Of course, this is only a concern if the API might actually return a result that is missing some of the possible keys we're looking for. Do you know if that is the case here?

Reply:
I don't think there's any case where the response omits keys.

The one exception, which is already being accounted for, are fields marked as "linux only" on the [api docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.16/cluster-nodes-stats.html). The only field we use that's linux only is io_stats, which we have as a pointer here, so we can differentiate between whether it was actually omitted or not.

Reply:
Great, thanks for explaining. Looks good then.